docs/src/pages/ui/theme/alternativeStyling/index.page.mdx

```diff
@@ -0,0 +1,501 @@
+---
+title: Alternative Styling
+---
+
+import Block from 'amplify-docs/src/components/Block';
+import BlockSwitcher from 'amplify-docs/src/components/BlockSwitcher';
+
+One of the goals of Amplify is to be a good citizen of the environment it exists in. As such Amplify works hard to not interfere with other tools that are being used.
+
+## CSS In JS
+
+There are many CSS In JS libraries out there that give a number of benefits. Amplify does not use a CSS In JS library to implement its theming system but operates well with the major CSS In JS libraries out there. Below are a few of the CSS In JS libraries out there and a description of how they interact with Amplify's theming system.
+
+<BlockSwitcher>
+<Block name="Styled Components">
+[Styled Components](https://github.com/styled-components/styled-components) allows you to create visual primitives with actual CSS code directly tied to components.
+
+#### Usage
+
+```jsx
+import React from 'react';
+import styled from 'styled-components';
+import { View } from '@aws-amplify/ui-react';
+
+const StyledView = styled(View)`
+  background-color: black;
+  font-size: 32px;
+`;
+
+//use it like any other amplify-ui component
+<StyledView color="red" className="my-styled-view">
+  This is my Styled View
+</StyledView>;
+```
+
+#### Interactions
+
+There are various ways to customize an Amplify component and Styled Components will interact with these customizations in the following ways:
+
+##### Amplify Styling Props
+
+These are styling props that can be used directly on an Amplify component which will affect a single style property. Examples of styling props are `color` or `fontWeight`. These Amplify styling props will take precedence over Styled Component styling.
+
+```jsx
+import React from 'react';
+import styled from 'styled-components';
+import { View } from '@aws-amplify/ui-react';
+
+const StyledView = styled(View)`
+  color: blue;
+`;
+
+<StyledView color="red">Using Styling props</StyledView>;
+```
+
+In the example above, the color of the view will be set to `red` because of the Amplify styling prop. [Here]() is a list of all Amplify styling props.
+
+##### Amplify variation props
+
+These are Amplify props that are added to certain Amplify components to change the look and/or behavior of the Amplify component. Examples of these variation props are `size` or `variation`.
+
+```jsx
+import React from 'react';
+import styled from "styled-components";
+import { Button } from '@aws-amplify/ui-react';
+
+const StyledButton = styled(Button)`
+  color: blue
+`;
+
+<StyledButton variation="primary'>Primary Button</StyledButton>
+```
+
+In the example above, the color of the button will be set to `white`, because the primary variation of the button has a color set to `white` and it takes precedence over the Styled Components styling. A list of Amplify variations can be found on [here]().
```

Comment:
Interesting, is this always the case or is it just because with variations our selector has a higher precedence? If you used a button without a variation would the color be blue?

Reply:
Yes this is because our variation styling rules have a higher precedence, without the variation on the button the library styling would take precedence over ours and the button would be color blue.

Reply:
Couldn't the customer use a data attribute selector to make it work, like this?
```
const StyledButton = styled(Button)`
  &[data-variation='primary']: {
    color: blue
   }
`;
```

Reply:
This makes me wonder if we should try to have the same specificity for all of our custom rules... The attribute selector rules certainly complicate things.

Reply:
Yea I'm thinking that as well. I guess that means we have to go with BEM or BEM-like syntax then?

Reply:
yes the developer can use data attribute selectors to make their rules take precedence. In this guide I was trying to describe the experience that they would run into if they simply use a CSS in JS library without putting in extra effort to work well with amplify.  It seems like there are variations and work arounds for anything they might want to do, but I was trying to not branch off into too many tangents for this guide.